libs/runtime/unit/platform_unit_test.cpp

```diff
@@ -607,6 +629,252 @@ TEST_CASE("epoch_test_stale_items", "[platform]")
+            step_t;
+
+        std::map<std::string, step_t> steps;
+
+        // Define possible steps the script can take.
+
+        // Initialize the state.
+        steps["setup"] = [&] {
+            work_items.clear();
+            for (auto i = 0; i < 2; i++) {
+                work_item_contexts[i].signal.reset();
+                auto work_item = ebpf_epoch_allocate_work_item(&work_item_contexts[i], work_item_context_t::invoke);
+                work_items.push_back(work_item_ptr(work_item, ebpf_epoch_cancel_work_item));
+            }
+        };
+
+        // Switch to running on CPU N
+        steps["switch_cpu"] = [&](size_t i) {
+            uintptr_t old_affinity;
+            (void)ebpf_set_current_thread_affinity(1ull << i, &old_affinity);
+        };
+
+        // Enter epoch for location N
+        steps["enter_epoch"] = [&](size_t i) { epoch_states[i].enter(); };
+
+        // Exit epoch for location N
+        steps["exit_epoch"] = [&](size_t i) { epoch_states[i].exit(); };
+
+        // Submit work item N
+        steps["schedule"] = [&](size_t i) { ebpf_epoch_schedule_work_item(work_items[i].release()); };
+
+        // Wait for work item N to complete with success or timeout.
+        steps["wait"] = [&](size_t i, bool expect_success) {
+            if (expect_success) {
+                work_item_contexts[i].signal.wait();
+            } else {
+                REQUIRE(!work_item_contexts[i].signal.wait_for(1s));
+            }
+        };
+
+        // Explicitly cleanup all state.
+        steps["cleanup"] = [&] {
+            for (auto i = 0; i < work_items.size(); i++) {
+                work_items[i].reset();
+            }
+        };
+
+        // Invoke epoch synchronize to ensure that all memory that is no longer in use is released.
+        steps["synchronize"] = [&] { ebpf_epoch_synchronize(); };
+
+        // Setup the state for the test.
+        std::get<std::function<void()>>(steps["setup"])();
+
+        // Execute each step in the script.
+        for (const auto& step : script) {
+            std::vector<std::string> tokens;
+            std::istringstream iss(step);
+
+            // Tokenize the step extracting the command and arguments.
+            for (std::string token; std::getline(iss, token, ',');) {
+                tokens.push_back(token);
+            }
+
+            // Verify that the command is valid.
+            REQUIRE(steps.find(tokens[0]) != steps.end());
+
+            try {
+                switch (tokens.size()) {
+                case 1:
+                    std::get<std::function<void()>>(steps[tokens[0]])();
+                    break;
+                case 2:
+                    std::get<std::function<void(size_t)>>(steps[tokens[0]])(std::stoi(tokens[1]));
+                    break;
+                case 3: {
+                    bool success = tokens[2] == "signalled";
```

Comment:
I know that this is incredibly picky, but I think that _signalled_ is _signaled_.

Reply:
Right, "signaled" is American spelling, "signalled" is British spelling.
This project uses American spelling.

Reply:
It doesn't help that my spell checker accepts both versions :-)